test/init.f90

```diff
@@ -10,19 +10,40 @@
 
 program hello
    use dlaf_fortran, only: dlaf_initialize, dlaf_finalize
+   use dlaf_fortran, only: dlaf_create_grid_from_blacs, dlaf_free_grid, dlaf_free_all_grids
    use testutils, only: setup_mpi, teardown_mpi
 
    implicit none
+   
+   external blacs_get
+   external blacs_gridinit
 
    integer :: rank, numprocs
    integer:: nprow, npcol
+   integer :: ictxt_0, ictxt_1
 
    nprow = 2
    npcol = 3
 
    call setup_mpi(nprow, npcol, rank, numprocs)
 
+   ! Get default system context
+   call blacs_get(0, 0, ictxt_0)
+   call blacs_get(0, 0, ictxt_1)
+
+   call blacs_gridinit(ictxt_0, 'R', nprow, npcol)
+   call blacs_gridinit(ictxt_1, 'R', nprow, npcol)
+
    call dlaf_initialize()
+
+   call dlaf_create_grid_from_blacs(ictxt_0)
+   call dlaf_create_grid_from_blacs(ictxt_1)
+   
+   call dlaf_free_grid(ictxt_0)
+
+   ! All grids need to be freed before MPI is finalized
+   call dlaf_free_all_grids()
```

Comment:
I suppose we don't have a way to check that `ictxt_1` was freed directly? Or will the test blow up at the end if it wasn't freed?

Reply:
We do not, because we can't interrogate the grid container directly. Do you think it would be useful to have?

I added it just to showcase both functions in this simple test, but `dlaf_free_all_grids` is enough. I could use only that. The test does fail if the grids are not freed, but `dlaf_free_all_grids` takes care of freeing all grids anyways.

Reply:
> We do not, because we can't interrogate the grid container directly. Do you think it would be useful to have?

I don't think so, just checking.
 
> The test does fail if the grids are not freed

This was what I mostly wanted to know! If it fails, without it, all good.

Reply:
Re-reading this, I think I confused `ictxt_1` you were asking about with `ictxt_0`. We don't have a way to know the grid related to `ictxt_0` was released, before all grids are released.

Since the test fails if grids are not released, I'm thinking it might be worth reverting the `init` test to just initialization/finalization, and then have two new separate tests for `dlaf_free_grid` and `dlaf_free_all_grids`.

Reply:
I took the approach described above.